src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h

```diff
@@ -1137,9 +1137,28 @@ ep_rt_entrypoint_assembly_name_get_utf8 (void)
 
 	// get the name from the host if we can't get assembly info, e.g., if the runtime is
 	// suspended before an assembly is loaded.
+	// We'll cache the value in a static function global as the caller expect the lifetime of this value
+	// to outlast the calling function.
+	static const ep_char8_t* entrypoint_assembly_name = NULL;
+	static bool read_name_from_host = false;
```

Comment:
I don't think this is needed. I would simply check if `entrypoint_assembly_name` is `NULL`. If `HOST_PROPERTY_ENTRY_ASSEMBLY_NAME` is `NULL`, allocate an empty string.

Reply:
I think null vs empty string would be an observable difference and I don't have the experience in the area to know if that change would cause any issues with tooling. I'll take a look and see if that difference would propagate through.

Reply:
They return an empty string on line 1164.